[Flake] TestWatchNotHangingOnStartupFailure

Observed in https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/131538/pull-kubernetes-unit/1917344434945527808

> {Failed  === RUN   TestWatchNotHangingOnStartupFailure
    cacher_whitebox_test.go:991: Unexpected error: &errors.StatusError{ErrStatus:v1.Status{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:"", RemainingItemCount:(*int64)(nil)}, Status:"Failure", Message:"storage is (re)initializing: failed to list *example.Pod: dummy", Reason:"TooManyRequests", Details:(*v1.StatusDetails)(0xc006bb40c0), Code:429}}
--- FAIL: TestWatchNotHangingOnStartupFailure (0.00s)


cc: @wojtek-t @serathius

--- a/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go
+++ b/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go
@@ -987,7 +987,7 @@ func TestWatchNotHangingOnStartupFailure(t *testing.T) {
 			t.Errorf("Unexpected error: %#v", err)
 		}
 	} else {
-		if err == nil || err.Error() != apierrors.NewTooManyRequests("storage is (re)initializing", 1).Error() {
+		if err == nil || !strings.Contains(err.Error(), "storage is (re)initializing") {
 			t.Errorf("Unexpected error: %#v", err)
 		}
 	}

